js/preview/src/build.ts

```diff
@@ -26,6 +26,22 @@ export async function make_build({
 			const pkg = JSON.parse(
 				fs.readFileSync(join(source_dir, "package.json"), "utf-8")
 			);
+			let component_config = {
+				plugins: [],
+				svelte: {
+					preprocess: []
+				}
+			};
+
+			if (
+				comp.frontend_dir &&
+				fs.existsSync(join(comp.frontend_dir, "gradio.config.js"))
+			) {
+				const m = await import(join(comp.frontend_dir, "gradio.config.js"));
```

Comment:
I went for `gradio.config.js` because that is a pretty solid convention in the JS world.

It has to be a `js` file rather than yaml/toml/json because these configs require you to run/ import code inside them. It also needs to be in the frontend dir because in the default case that is the dir with access to node modules.

This is a little complicated though because you could multiple frontend dirs. We can technically support a config per component for the build as each component build is a discreet process.

Reply:
Should we create an empty `gradio.config.js` file during `gradio cc create`?

Reply:
Yeah I was thinking about that. Not sure. It's helpful but also adds noise.